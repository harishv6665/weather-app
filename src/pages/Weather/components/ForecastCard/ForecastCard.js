import React from "react";
import moment from "moment";
import PropTypes from "prop-types";
import Card from "../../../../shared/components/Card/Card";
import styles from "./ForecastCard.module.css";
import temperatureConverter from "../../../../shared/utils/temperatureConverter";

const ForecastCard = ({ data, isActive, onCardClick, isCelsius }) => (
    <Card customStyles={{ card: `${styles.card} ${isActive ? styles.active: ""}`}} onClick={onCardClick}>
        <React.Fragment>
            <div className={styles.card__field}>
                <span className={styles.card__field__label}>Date: </span>
                <span className={styles.card__field__value}>{moment(data.date).format("Do MMM YY")}</span>
            </div>
            <div className={styles.card__field}>
                <span className={styles.card__field__label}>Temp: </span>
                <span className={styles.card__field__value}>
                    {temperatureConverter(data.avg_temp, isCelsius)}
                    {isCelsius ? " ºC" : " ºF"}
                </span>
            </div>
        </React.Fragment>
    </Card>
)

ForecastCard.propTypes = {
    data: PropTypes.object,
    isActive: PropTypes.bool,
    onCardClick: PropTypes.func,
    isCelsius: PropTypes.bool
}

ForecastCard.defaultProps = {
    data: {},
    isActive: false,
    onCardClick: () => {},
    isCelsius: true
}

export default ForecastCard;
