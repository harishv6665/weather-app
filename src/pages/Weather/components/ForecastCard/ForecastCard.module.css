.card {
    width: 100%;
    padding: 16px 24px;
    margin-bottom: 16px;
    display: flex;
    align-items: center;
    cursor: pointer;
    justify-content: space-between;
}

.card:last-child {
    margin-bottom: 0;
    margin-right: 0;
}

.card.active {
    border: 1px solid var(--themeBg);
    box-shadow: 6px 6px 12px #c5c5c5, -6px -6px 12px #ffffff;
}

.card__field {
    display: flex;
    flex-direction: column;
}
.card__field__label {
    font-size: 12px;
    color: var(--secondaryText);
}

.card__field__value {
    font-size: 20px;
    font-weight: 100;
    padding-top: 4px;
}

@media only screen and (min-width: 768px) {
    .card {
        width: calc((100% / 3) - 8px);
        height: auto;
        margin-right: 16px;
        margin-bottom: 0;
        flex-direction: column;
        align-items: start;
    }

    .card__field {
        margin-bottom: 16px;
    }

    .card__field:last-child {
        margin-bottom: 0;
    }
}
