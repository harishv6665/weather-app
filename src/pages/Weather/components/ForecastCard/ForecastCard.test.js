import React from "react";
import { shallow } from "enzyme";
import ForecastCard from "./ForecastCard";

const CARD_DATA = {};
const CARD_IS_ACTIVE = false;
const CARD_IS_CELSIUS = false;

describe("ForecastCard tests", () => {

    it("should be defined", () => {
        expect(ForecastCard).toBeDefined();
    });

    it("should match to snapshot", () => {
        const wrapper = shallow(<ForecastCard data={CARD_DATA} isActive={CARD_IS_ACTIVE} isCelsius={CARD_IS_CELSIUS} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should match the DOM structure", () => {
        const wrapper = shallow(<ForecastCard data={CARD_DATA} isActive={CARD_IS_ACTIVE} isCelsius={CARD_IS_CELSIUS} />);
        expect(wrapper.find("Card").length).toBe(1);
        expect(wrapper.find("ForecastCardField").length).toBe(2);
        expect(wrapper.find("Card ForecastCardField").length).toBe(2);
    });

    it('should simulate on click', () => {
        const mockFn = jest.fn();
        const wrapper = shallow(<ForecastCard
            data={CARD_DATA}
            isActive={CARD_IS_ACTIVE}
            isCelsius={CARD_IS_CELSIUS}
            onCardClick={mockFn}
        />);
        wrapper.simulate('click');
        expect(mockFn).toHaveBeenCalled();
    });

});
