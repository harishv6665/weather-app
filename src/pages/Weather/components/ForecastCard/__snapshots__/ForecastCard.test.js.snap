// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastCard tests should match to snapshot 1`] = `ShallowWrapper {}`;
